src/containers/habit-tracker/analytics/Analytics.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect } from "react";
+import styled, { keyframes } from "styled-components";
+import { PageBackground } from "../../../components/PageBackground";
+import { ToggleButton } from "../../../components/habit-tracker/analytics/ToggleButton";
+import { MonthView } from "./month-view/MonthView";
+import { YearView } from "./year-view/YearView";
+import { Habit, Logging } from "../habits/Habits";
+import { LocalStorage } from "../../../common/constants/local-storage";
+
+// Enum for view modes
+enum ViewMode {
+  MONTH = "month",
+  YEAR = "year",
+}
+
+const fadeUp = keyframes`
+  from {
+    opacity: 0;
+    transform: translateY(20px);
+  }
+  to {
+    opacity: 1;
+    transform: translateY(0);
+  }
+`;
+
+// Styled component for the main container of the analytics page
+const AnalyticsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: calc(100% - 190px);
+  height: calc(100vh - 56px);
+  overflow-y: auto;
+  position: absolute;
+  top: 56px;
+  left: 190px;
+  padding: 20px;
+  box-sizing: border-box;
+  animation: ${fadeUp} 1s ease-out;
+
+  @media (max-width: 768px) {
+    width: calc(100% - 70px);
+    left: 70px;
+  }
+`;
+
+const toggleOptions = [
+  { label: "Month View", value: ViewMode.MONTH },
+  { label: "Year View", value: ViewMode.YEAR },
+];
+
+// The main functional component for the Analytics page of the habit tracker
+export function Analytics() {
+  // State to manage the current view (month or year)
+  const [view, setView] = useState<ViewMode>(ViewMode.MONTH);
+  const [habits, setHabits] = useState<Habit[]>([]);
+  const [habitsLog, setHabitsLog] = useState<Logging>({});
+  const [selectedHabit, setSelectedHabit] = useState<string>("");
+  const [selectedYear, setSelectedYear] = useState<number>(
```

Comment:
we are not using this state in this component. Better to move it to the YearView component.